refactor(transformer): introduce `TransformCtx::duplicate_expression`

overlookmotel opened the pull request:
Move implementation of `duplicate_object` from the class properties transform into `TransformCtx::duplicate_expression`, so it can also be used in other transforms.

overlookmotel commented:
* **#7745** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7745?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7757** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7757?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7756** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7756?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7755** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7755?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7754** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7754?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7754?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
Great! It also can be used in `optional-chaining` plugin!

overlookmotel committed:
refactor(transformer): introduce `TransformCtx::duplicate_expression` (#7754)

Move implementation of `duplicate_object` from the class properties transform into `TransformCtx::duplicate_expression`, so it can also be used in other transforms.
